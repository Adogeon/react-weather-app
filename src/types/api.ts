import type { CurrentWeather, DailyForecastWithHour } from "./weather";

export interface rawAPIError {
    error: {
        code: string;
        message: string;
    };
}

export interface rawAPISuccess {
    location: {
        name: string;
        region: string;
        country: string;
    };
    current: CurrentWeather
    forecast: {
        forecastday: Array<DailyForecastWithHour>
    }
};
