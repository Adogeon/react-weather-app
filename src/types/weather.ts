export interface WeatherCondition {
    text: string;
    icon: string;
}

export interface TempAndConditions {
    temp_c: string;
    temp_f: string;
    feelslike_c: string;
    feelslike_f: string;
    condition: WeatherCondition;
}

export interface CurrentWeather extends TempAndConditions {
    last_updated: string
}

interface DailyForecastWeather {
    maxtemp_c: string;
    maxtemp_f: string;
    mintemp_c: string;
    mintemp_f: string;
    daily_chance_of_rain: string;
    avghumidity: string;
    condition: WeatherCondition;
}

export interface DailyForecastData {
    date: string;
    day: DailyForecastWeather
}

interface HourlyForecast extends TempAndConditions {
    time: string;
}

export interface DailyForecastWithHour extends DailyForecastData {
    hour: Array<HourlyForecast>
}