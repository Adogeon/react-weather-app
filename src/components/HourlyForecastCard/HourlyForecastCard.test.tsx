import { test, expect } from "vitest";
import { render } from "@testing-library/react";
import HourlyForcastCard from ".";
import { HourlyForecast } from "../../types/weather";

const testData: HourlyForecast = {
  time: "2024-11-15 14:00",
  temp_c: "14.3",
  temp_f: "57.7",
  feelslike_c: "12.7",
  feelslike_f: "54.9",
  condition: {
    text: "test",
    icon: "/64x64/test",
  },
};

test("HourlyForcastCard should render correctly", () => {
  const { asFragment } = render(<HourlyForcastCard data={testData} />);

  expect(asFragment()).toMatchSnapshot();
});
