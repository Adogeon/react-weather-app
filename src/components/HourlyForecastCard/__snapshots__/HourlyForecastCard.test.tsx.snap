// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HourlyForcastCard should render correctly 1`] = `
<DocumentFragment>
  <div
    class="_hourly-card_0df995"
  >
    <div>
      14 PM
    </div>
    <div>
      <img
        alt="test-icon"
        src="/64x64/test"
      />
    </div>
    <div>
      14.3Â°C
    </div>
  </div>
</DocumentFragment>
`;
